teuthology/task/install/__init__.py

```diff
@@ -565,16 +566,24 @@ def task(ctx, config):
     log.debug('project %s' % project)
     overrides = ctx.config.get('overrides')
     repos = None
-    extra_system_packages = config.get('extra_system_packages', [])
 
     if overrides:
-        install_overrides = overrides.get('install', {})
-        log.debug('INSTALL overrides: %s' % install_overrides)
-        teuthology.deep_merge(config, install_overrides.get(project, {}))
-        overrides_extra_system_packages = install_overrides.get('extra_system_packages')
-        if overrides_extra_system_packages:
-            teuthology.deep_merge(extra_system_packages, overrides_extra_system_packages)
-        repos = install_overrides.get('repos', None)
+        try:
+            install_overrides = overrides.get('install', {})
+            log.debug('INSTALL overrides: %s' % install_overrides)
+            teuthology.deep_merge(config, install_overrides.get(project, {}))
+            overrides_extra_system_packages = install_overrides.get('extra_system_packages')
+            if overrides_extra_system_packages:
+                extra_system_packages = config.get('extra_system_packages')
+                config['extra_system_packages'] = teuthology.deep_merge(extra_system_packages, overrides_extra_system_packages)
```

Comment:
```suggestion
                teuthology.deep_merge(extra_system_packages, overrides_extra_system_packages)
```

? (This one confused me.)

Reply:
The result of the deep merge should be stored in `config` because later we do [this](https://github.com/ceph/teuthology/pull/1942/files#diff-28d5292d55b79ee4e4e70922a033dbec3a489da3fb4f6672ad8fb64ab053b84cR621) `extra_system_packages=config.get('extra_system_packages', []),`

Using `config.get('extra_system_packages')` is better than directly using variable `extra_system_packages` because when [we log whole config](https://github.com/ceph/teuthology/pull/1942/files#diff-28d5292d55b79ee4e4e70922a033dbec3a489da3fb4f6672ad8fb64ab053b84cR588) there would be no doubt what the actual config would look like.

Reply:
I think it'll only be the same if `config.get('extra_system_packages')` is a list or a dictionary. If it's `None` (case:`extra_system_packages` not mentioned in install task definition and only in overrides), then `extra_system_packages` wouldn't reference it. 

I think the regression (https://tracker.ceph.com/issues/66133) happened because of the above case:
"python3-pytest" package was defined in `overrides.install.ceph.extra_system_packages` and there was no `extra_system_packages` in install task definition config. So the variable was an empty list which didn't reference the config value. 
Then `teuthology.deep_merge(config, install_overrides.get(project, {}))` added the package to `config["extra_system_packages"]` (but not in `extra_system_packages` variable). We could see the package when config was logged but since we were [using extra_system_package variable](https://github.com/ceph/teuthology/pull/1943/files#diff-28d5292d55b79ee4e4e70922a033dbec3a489da3fb4f6672ad8fb64ab053b84cR612) to makeup the final config, the package never got added to final config and so it was not installed.